db/init_db.go

```diff
@@ -27,6 +28,13 @@ func NewDB() (*DB, error) {
 
 	return &DB{Queries: q, pool: pool}, nil
 }
+func (db *DB) NewTx(ctx context.Context) (pgx.Tx, error) {
+	tx, err := db.pool.Begin(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return tx, nil
```

Comment:
```suggestion
	return db.pool.Begin(ctx)
```
Upon inspecting the source code I noticed `pool.Begin()` already returns `nil` when the operation fails